lib/upgrade/etcd/conversionv1v3/ippool.go

```diff
@@ -21,31 +21,61 @@ import (
 	log "github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
 	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	"github.com/projectcalico/libcalico-go/lib/ipip"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
 )
 
-func ConvertIPPool(kvp *model.KVPair) (*model.KVPair, error) {
-	pool, ok := kvp.Value.(model.IPPool)
+// ippool implements the Converter interface.
+type ippool struct{}
```

Comment:
Probably want to make this public, e.g. IPPoolConverter

Reply:
Oooh, actually I think you'll want the methods to be public too.  I suspect the upgrade code will not be written in this package.

Reply:
yes good point, I'll export them